chain/network/src/routing/routing_table_actor.rs

```diff
@@ -211,14 +238,10 @@ impl RoutingTableActor {
                                     ColPeerComponent,
                                     peer_id.try_to_vec().unwrap().as_ref(),
                                 );
-                            } else {
-                                warn!("We expected `peer_id` to belong to component {}, but it belongs to {}",
-                                       component_nonce, cur_nonce);
                             }
-                        } else {
-                            warn!("We expected `peer_id` to belong to a component {}, but it doesn't belong anywhere",
-                                       component_nonce);
+                            // else case can happen, see explanation of this function.
```

Comment:
based on the explanation above, we don't have to print the warning if cur_nonce > component_nonce (as this is the case that you've described in explanation).

But what if cur_nonce > component_nonce ? 

let's discuss it during our meeting today.

Reply:
@mm-near That's completely normal. 



Case A) You add components C1, then C2.
Then you remove then in order C2, then C1.

Case B) You add components C1, then C2.
Then you remove then in order C1, then C2.

There is nothing special with `cur_nonce < component_nonce` vs  `cur_nonce > component_nonce`

Reply:
There are some advanced cases involving multiple cliques that I didn't cover. I think it's not worth documenting them.